drivers/clock_control/CMakeLists.txt

```diff
@@ -9,6 +9,7 @@ zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_MCUX_SIM            clock_control_mcux
 zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_NRF                 nrf_power_clock.c)
 zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION nrf_clock_calibration.c)
 zephyr_sources_ifdef(CONFIG_CLOCK_CONTROL_RV32M1_PCC          clock_control_rv32m1_pcc.c)
+zephyr_sources_ifdef(CONFIG_SOC_ESP32          clock_control_esp32.c)
```

Comment:
Since everything else have `CONFIG_CLOCK_CONTROL_` as prefix, maybe add a new Kconfig for this, and force `select` under `CONFIG_SOC_ESP32`? Not too sure about the policy here.

Reply:
I want to enforce the driver, because without it, all peripherals will fail to work.
I'm not sure if I can enforce the driver using `CONFIG_CLOCK_CONTROL_`, and don't display an option to disable it from menuconfig

Reply:
You can use the `select` keyword. In `soc/xtensa/esp32/Kconfig.soc`, `SOC_ESP32` "selects" `XTENSA` so that it cannot be disabled.

Reply:
I added
```
	select CLOCK_CONTROL
	select CLOCK_CONTROL_ESP32
```
Without `CLOCK_CONTROL` you can just disable all clock drivers, which will lead to the same issue.